src/bus/mod.rs

```diff
@@ -36,7 +34,7 @@ pub struct Bus {
 // All (cheaply) cloneable fields of `Bus` go here.
 #[derive(Clone, Debug)]
 pub struct Inner {
-    address: Address,
+    address: String,
```

Comment:
This is not acceptable to me. We should move from generic strings to specific types, not the other way around.

Reply:
hmm.. I'm not sure I want that. We should only support 1 address at a time. I want to keep things simpler. Unless there is a compelling use case for this?

Reply:
well, I have some use cases in mind, like allowing the guest services to connect to a host VM bus (think qemu / spice agents etc)

Reply:
That sounds good at first but giving raw direct access to the host's system bus would be a bad idea, security wise. Even for session, it can be argued that it's not good. There is a reason why xdg-dbus-proxy exists since people don't even want apps running on the same host, unfiltered access to the bus. So I wonder if there could better solutions for this, especially ones that don't require listening to multiple connections.

Also to keep in mind that you wouldn't want users to have to modify the dbus configuration but rather make everything work out of the box.

Reply:
In any case, let's look into enabling multiple sockets once we get to adding use cases that would require it. For now, let's keep it simple.

Reply:
no, it is a private bus, as started by libvirt on your behalf when using display dbus, or other helpers, and soon an rdp server (I am slowly working on that too).

Reply:
> no, it is a private bus

That's cool, but then why does it need to listen on multiple addresses?

Reply:
so the guest services can connect to the host bus, and management apps can use regular unix socket.

Reply:
> so the guest services can connect to the host bus

You mean through vsock? I am not sure why qemu does it differently but the vsock impl of Firecracker's way of using a plan unix socket on the host side, makes a lot of sense to me. That will also solve this issue.

Reply:
You need some kind of vsock<->unix bridge for that, and you loose direct peer details.

Reply:
> You need some kind of vsock<->unix bridge for that, and you loose direct peer details.

like peer credentials? I thought that's not possible and why we use anonymous auth with vsock. :thinking: